earthfile2llb/listener.go

```diff
@@ -1091,3 +1100,69 @@ func unescapeSlashPlus(str string) string {
 	// TODO: This is not entirely correct in a string like "\\\\+".
 	return strings.ReplaceAll(str, "\\+", "+")
 }
+
+type matrixEntry struct {
+	key    string
+	values []*string
+}
```

Comment:
Isnt this just a long way to say `type matrixEntry map[string][]*string`? Just nitpicking; but would be fairly clear.

Reply:
Yes - with one very important distinction: the order in a map is not guaranteed. Different ordering on each run can mean unnecessary cache-busting - that's why it's all slices and structs.